lib/api.js

```diff
@@ -160,10 +163,14 @@ function NodeCG(bundle, socket) {
 
 		socket.on('error', err => {
 			if (err.type === 'UnauthorizedError') {
-				const url = [location.protocol, '//', location.host, location.pathname].join('');
-				window.location.href = `/authError?code=${err.code}&message=${err.message}&viewUrl=${url}`;
+				if (globalThis.window) {
+					const url = [location.protocol, '//', location.host, location.pathname].join('');
+					window.location.href = `/authError?code=${err.code}&message=${err.message}&viewUrl=${url}`;
+				} else {
+					globalThis.close();
```

Comment:
Is it guaranteed that `close` is available in all contexts where `globalThis.window` isn't set?

Reply:
The only three browser scopes where globalThis.window is not defined are the three types of WorkerGlobalScope. The close method is available in DedicatedWorkerGlobalScope and SharedWorkerGlobalScope. The ServiceWorkerGlobalScope doesn't have a close method, however a Service Worker cannot create a WebSocket connection, so NodeCG would not properly work in a Service Worker in the first place.

So in all cases where the code would actually be able to reach this point, the close method is available.